### PR TITLE
Stabilize 'entry_and_modify' feature

### LINKED ISSUE #44733: Tracking issue for `entry_and_modify`
Tracking issue for the `and_modify` method on hash map `Entry`s (added in https://github.com/rust-lang/rust/pull/44734).

### PR DESCRIPTION
**Pull request by hedgehog1024**
Stabilize `entry_and_modify` feature introduced by #44734.

Closes #44733

### CODE REVIEW

**Comment by cramertj** [comment]
r? @sfackler


(Or someone else on libs)

**Comment by sfackler** [comment]
@rfcbot fcp merge

File: src/libstd/collections/hash/map.rs [hunk 1]
```diff
@@ -2081,7 +2080,7 @@ impl<'a, K, V> Entry<'a, K, V> {
     ///    .or_insert(42);
     /// assert_eq!(map["poneyland"], 43);
     /// ```
-    #[unstable(feature = "entry_and_modify", issue = "44733")]
+    #[stable(feature = "entry_and_modify", since = "1.25.0")]
```
Comment (kennytm) [inline src/libstd/collections/hash/map.rs]: The version number should be changed to 1.26.0

**Comment by Manishearth** [comment]
(Fixed in an extra commit)

File: src/liballoc/btree/map.rs [hunk 2]
```diff
@@ -2129,7 +2128,7 @@ impl<'a, K: Ord, V> Entry<'a, K, V> {
     ///    .or_insert(42);
     /// assert_eq!(map["poneyland"], 43);
     /// ```
-    #[unstable(feature = "entry_and_modify", issue = "44733")]
+    #[stable(feature = "entry_and_modify", since = "1.25.0")]
```
Comment (kennytm) [inline src/liballoc/btree/map.rs]: @Manishearth This also needs to be 1.26.0...

**Pull request merged**